packages/ai/src/registry/custom-provider.ts

```diff
@@ -41,8 +41,8 @@ export function customProvider<
   imageModels?: IMAGE_MODELS;
   transcriptionModels?: TRANSCRIPTION_MODELS;
   speechModels?: SPEECH_MODELS;
-  fallbackProvider?: ProviderV2;
-}): ProviderV2 & {
+  fallbackProvider?: ProviderV3;
```

Comment:
we may want to introduce `Provider = ProviderV2 | ProviderV3` to allow old community providers to still be used here